bluebubbles-server/src/server/databases/imessage/listeners/outgoingMessageListener.ts

```diff
@@ -128,29 +133,25 @@ export class OutgoingMessageListener extends ChangeListener {
             const cacheName = getCacheName(entry);
 
             // Skip over any that we've finished
-            if (this.cache.find(cacheName)) return;
+            if (this.cache.find(cacheName)) continue;
 
             // Add to cache
             this.cache.add(cacheName);
 
             // Resolve the promise for sent messages from a client
             const idx = Server().messageManager.findIndex(entry);
             if (idx >= 0) {
-                Server().messageManager.promises[idx].resolve(entry);
-
-                // If we have a message match, we want the message match event to
-                // get to the clients first, so I'm adding an artificial delay.
-                // This _only_ applies when a message match is found, meaning you
-                // sent it from a client
-                setTimeout(() => {
-                    super.emit("new-entry", entry);
-                }, 1000);
+                await Server().messageManager.promises[idx].resolve(entry);
+                // After the MessageMatchEmit event reaches client, then emmit new-entry to fcm
+
```

Comment:
Sorry for the nitpick, but mind removing the extra returns here. And put the comment before the resolve line